packages/pages/src/components/analytics/Analytics.ts

```diff
@@ -0,0 +1,231 @@
+import { TemplateProps } from "../../common/src/template/types";
+import { getRuntime, isProduction } from "../../util";
+import { AnalyticsMethods } from "./interfaces";
+import {
+  ConversionDetails,
+  CookieManager,
+  EntityPage,
+  PagesAnalyticsService,
+  providePagesAnalytics,
+  StaticPage,
+  Visitor,
+} from "@yext/analytics";
+import { concatScopes, slugify } from "./helpers";
+import { useScope } from "./scope";
+
+/**
+ * The Analytics class creates a stateful facade in front of the \@yext/analytics
+ * Library's pagesAnalyticsProvider class. It takes in some data from the
+ * template configuration and uses it to provide configuration to the
+ * pagesAnalyticsProvider.
+ *
+ * Additionally, it provides handlers for controlling user opt-in for compliance
+ * requirements as well as for debugging, enabling conversion tracking, saving
+ * user identity information, and creating named analytics scopes for
+ * easy tagging.
+ *
+ * @public
+ */
+export class Analytics implements AnalyticsMethods {
+  private _optedIn: boolean;
+  private _conversionTrackingEnabled = false;
+  private _cookieManager: CookieManager | undefined;
+  private _analyticsReporter: PagesAnalyticsService | undefined;
+  private _pageViewFired = false;
+  private _enableDebugging = false;
+  constructor(
+    private templateData: TemplateProps,
+    requireOptIn?: boolean | undefined
+  ) {
+    this._optedIn = !requireOptIn;
+    this.makeReporter();
+    this.pageView();
+  }
+
+  private calculatePageType(): EntityPage | StaticPage {
+    const isStaticPage = !!this.templateData.document?.__?.staticPage;
+    const isEntityPage = !!this.templateData.document?.__?.entityPageSet;
+
+    let pageType: EntityPage | StaticPage;
+
+    if (isStaticPage) {
+      pageType = {
+        name: "static",
+        staticPageId: this.templateData.document.__.name as string,
+      };
+    } else if (isEntityPage) {
+      pageType = {
+        name: "entity",
+        pageSetId: this.templateData.document.__.name as string,
+        id: this.templateData.document.uid as number,
+      };
+    } else {
+      throw new Error("invalid document type");
+    }
+
+    return pageType;
+  }
+
+  private makeReporter() {
+    if (getRuntime().name !== "browser") {
+      return;
+    }
+    if (!this._optedIn) {
+      return;
+    }
+
+    const inProduction =
+      isProduction(this.templateData?.document?.siteInternalHostName) ||
```

Comment:
Is the `siteInternalHostName` only populated in prod?

Reply:
the isProduction method returns true if window.location.href matches the domain name you pass in and you're in the browser environment